src/main.cpp

```diff
@@ -1,101 +1,140 @@
+	bool shouldDescribe;
+	bool shouldRun;
+	bool shouldShowHelp;
+	std::string filename;
+};
+
+CommandState parse_commands(const std::vector<std::string>& commands);
+void show_commands();
+
+std::string getName();
+CommandState whatShouldDo();
+
+void describe(ClassLoader& cl);
+void run(ClassLoader& cl, const std::string& filename);
+
 int main (int argc, char *argv[ ]) {
-	std::string show;
 
 	std::vector<std::string> commands(argv + 1, argv + argc);
 
 	try {
-		bool isToDescribe = false;
-		bool isToRun = false;
-		bool isHelp = false;
-		bool hasName = false;
-		std::string filename;
-
-		for (auto &command : commands) {
-			if (command == "--describe" || command == "-d") {
-				isToDescribe = true;
-			} else if (command == "--execute" || command == "-r") {
-				isToRun = true;
-			} else if (command == "--help"||  command == "-h") {
-				isHelp = true;
-			} else if (!hasName) {
-				filename = command;
-				hasName = true;
-			} else {
-				throw JvmException("Tem algum problema com os argumentos");
-			}
-		}
-
-		jvm::ClassLoader cl;
+		CommandState state = parse_commands(commands);
 
-		if (isHelp) {
+		if (state.shouldShowHelp) {
 			show_commands();
 			return 0;
 		}
 
-		if (!hasName) {
-			std::cout << "Digite o nome do arquivo: ";
-			std::cin >> filename;
-		}
+		if (state.filename.empty())
+			state.filename = getName();
 
-		cl.read(filename);
-
-		if (!isToDescribe && !isToRun) {
-			int op;
-
-			std::cout << "Digite 1 se deseja que o(s) .class seja(m) lido(s)" << std::endl;
-			std::cout << "Digite 2 se deseja que seja rodado o código" << std::endl;
-			std::cout << "Digite 3 se ambos" << std::endl;
-			std::cout << ": ";
-
-			std::cin >> op;
-
-			switch (op) {
-				case 1:
-					isToDescribe = true;
-					break;
-				case 2:
-					isToRun = true;
-					break;
-				case 3:
-					isToDescribe = true;
-					isToRun = true;
-					break;
-				default:
-					throw JvmException("Cara... Você tinha um trabalho, escolher entre 3 números...");
-			}
-		}
+		jvm::ClassLoader cl;
+		cl.read(state.filename);
 
-		if (isToDescribe) {
-			cl.show();
-		}
+		if (!state.shouldDescribe && !state.shouldRun)
+			state = whatShouldDo();
 
-		if (isToRun) {
-			jvm::Engine engine(cl);
-			auto index = filename.find_last_of("/\\");
-			engine.path = filename.substr(0, index + 1);
-			engine.execute();
-		}
+		if (state.shouldDescribe)
+			describe(cl);
 
-	} catch (std::string& error) {
+		if (state.shouldRun)
+			run(cl, state.filename);
+
+	} catch (const std::string& error) {
 	    std::cout << error << std::endl;
-	} catch (char* error) {
+	} catch (const char* const error) {
 		std::cout << error << std::endl;
-	} catch (jvm::JvmException& e) {
+	} catch (const jvm::JvmException& e) {
 		std::cout << e.what() << std::endl;
 	}
 
 	return 0;
 }
+
+CommandState parse_commands(const std::vector<std::string>& commands) {
+	CommandState state;
+	for (const auto &command : commands) {
+		if (command == "--describe" || command == "-d")
```

Comment:
@claudiosegala Ugh, NEVER!!!

Jokes aside, it was already like that when I found it. Also, I don't remember seeing anything using 'or' and 'and'. Wouldn't this change break uniformity? Perhaps, creating a new issue to address this would be better...